### ISSUE TITLE
compute example not running

### ISSUE THREAD

**Issue by hydroper** [gfx-rs/gfx#2355]
Short info header:
- GFX version: 0.1.0
- OS: Windows 10

```bat
> cargo run --bin compute --features dx12 1 2 3 4
    Finished dev [unoptimized + debuginfo] target(s) in 0.37s
     Running `D:\home\hydroper\Sources\RustLibs\gfx\target\debug\compute.exe 1 2 3 4`
ERROR 2018-08-28T15:12:30Z: gfx_backend_dx12: Failed on dxgi factory creation: -2005270527
thread 'main' panicked at 'assertion failed: !ptr.is_null()', C:\Users\neide\.cargo\registry\src\github.com-1ecc6299db9ec823\wio-0.2.0\src\com.rs:20:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: process didn't exit successfully: `D:\home\hydroper\Sources\RustLibs\gfx\target\debug\compute.exe 1 2 3 4` (exit code: 101)
```

**Comment by kvark** [comment]
Thanks for the report! Hopefully, someone with Windows10 can take a look at it shortly. In the meantime, you could help (unless you want to look at it yourself?) by setting `RUST_BACKTRACE=1` and providing us with the stack trace.

**Comment by hydroper** [comment]
@kvark Finally I figured out it's a environment variable.

Hm, this error looks to be within the `wio` dependency of gfx.

**Comment by msiglreith** [comment]
Does your gpu support dx12?

**Comment by hydroper** [comment]
@msiglreith According to

![image](https://user-images.githubusercontent.com/15965072/44740793-b4065200-aad1-11e8-9326-60c1869573ac.png)

Yes. I do also've the D3D12.dll in System32.

**Comment by msiglreith** [comment]
Thanks, it's actually on of the very first calls `CreateDXGIFactory2` failing.

* Does `quad` also crash?
* Could you try in release mode?
* Does the debug console (running in visual studio) emit something?

The next step would probably be then to isolate the [call](https://github.com/gfx-rs/gfx/blob/master/src/backend/dx12/src/lib.rs#L673) by removing the prior `debug_assertion` part and remove the dxgi1_3::DXGI_CREATE_FACTORY_DEBUG` flag.

Difficult to debug as I'm unable to reproduce it.

**Comment by hydroper** [comment]
@msiglreith 

- I just ran `quad` and it has also crashed the same way, with the same `wio` panic!
- I've again attempted to ran `quad` using `--release`, and the same panic occurred. Same call stack.
- I'm using MinGW (`x86_64-pc-windows-gnu`) to build `gfx` since VS is kinda expensive for my current internet.

> The next step would probably be then to isolate the call by removing the prior debug_assertion part and remove the dxgi1_3::DXGI_CREATE_FACTORY_DEBUG` flag.

I'll basically have to fork `wio` to change `dxgi1_3`, right? Hm, will give some tries. Ty for the tips!

**Comment by msiglreith** [comment]
@hydroper thanks for testing to so far

> I'm using MinGW (x86_64-pc-windows-gnu) to build gfx since VS is kinda expensive for my current internet.

Actually I'm not sure if MinGW supports dx12 so far. Not sure if someone has tested it so far.

> I'll basically have to fork wio to change dxgi1_3, right?

It just crashes in wio as we pass a nullptr. You could clone gfx repository and change the lines mentioned. The wio dependency can stay as is.

**Comment by hydroper** [comment]
@msiglreith Yeah, it was quite easy to change the lines. I stripped lines 673..682

```
        let hr = unsafe {
            dxgi1_3::CreateDXGIFactory2(
                dxgi1_3::DXGI_CREATE_FACTORY_DEBUG,
                &dxgi1_4::IDXGIFactory4::uuidof(),
                &mut dxgi_factory as *mut *mut _ as *mut *mut _)
        };

        if !winerror::SUCCEEDED(hr) {
            error!("Failed on dxgi factory creation: {:?}", hr);
        }
```

Then saved and ran `cargo run --bin compute --features dx12 1 2 3 4` on `examples` and still got the same result.

> Actually I'm not sure if MinGW supports dx12 so far. Not sure if someone has tested it so far.

That means MinGW could be incompatible with DXGI/`wio`. Well... I should sacrifice my internet for VS the 3rd time...

**Comment by kvark** [comment]
Relevant: https://github.com/gfx-rs/gfx/blob/ec67cfea199601159447d78734694132f5cfed09/Makefile#L16

**Comment by hydroper** [comment]
@kvark I've installed VS here and it worked. It was really looking like MinGW (GCC) wasn't supported. But I still don't understand 1 thing: is it normal that we can't open the `.exe` manually from the  `~/target/[debug|release]` folder? I get a `unwrap()` error if I try, but I can do `cargo run` fine.
